evm/src/memory/memory_stark.rs

```diff
@@ -304,14 +315,16 @@ impl<F: RichField + Extendable<D>, const D: usize> MemoryStark<F, D> {
             &segment_first_change,
             &virtual_first_change,
         );
+        let to_pad_to = max_timestamp_diff.next_power_of_two().max(num_trace_rows);
```

Comment:
I think we may need to use `max_timestamp_diff + 1`, to account for zero. Like if the max diff is `d`, we want to support `0..=d` as range check inputs, which is `d + 1` values.

Reply:
Also is `num_trace_rows` guaranteed to be a power of two here? If not we could move `next_power_of_two()` after the `max`

Reply:
Oh right!